### PR TITLE
Use eth_chainId instead of net_version for when guessing chain ID

### LINKED ISSUE #122: Use `eth_chainId` instead of `net_version` for guessing the chain ID.
`eth_chainId` is now widely supported and there is no sense in using the `net_version` (network ID) as the best guess for a chain ID.

### PR DESCRIPTION
**Pull request by nlordell**
As description. Note that this closes #122 

### Test Plan

CI

### CODE REVIEW

File: src/transaction.rs [hunk 1]
```diff
@@ -475,19 +475,14 @@ impl<T: Transport> BuildFuture<T> {
...
-                // like mainnet and various testnets and provides better safety
-                // against replay attacks then just using no chain ID; so lets
-                // reproduce that behaviour here
-                // TODO(nlordell): don't convert to and from string here
-                let chain_id = maybe!(chain_id.map(|id| id.to_string()), net.version());
+                let gas_price = maybe!(builder.gas_price, eth.gas_price().compat());
+                let nonce = maybe!(builder.nonce, eth.transaction_count(from, None).compat());
+                let chain_id = maybe!(
+                    chain_id.map(U64::from),
+                    CallFuture::new(transport.execute("eth_chainId", vec![])).compat()
```
Comment (bh2smith) [inline src/transaction.rs:484]: Is this a feature released with the recent Istanbul update?

**Review by bh2smith — APPROVED** [review]
Sweet deal. Very nice.

File: src/transaction.rs (hunk 1 above)
Comment (nlordell) [reply src/transaction.rs:484]: https://eips.ethereum.org/EIPS/eip-695

File: src/transaction.rs [hunk 2]
```diff
@@ -766,7 +759,7 @@ mod tests {
         transport.add_response(json!(gas));
         transport.add_response(json!(gas_price));
         transport.add_response(json!(nonce));
-        transport.add_response(json!(format!("{}", chain_id))); // chain id
+        transport.add_response(json!(format!("0x{:x}", chain_id)));
```
Comment (fleupold) [inline src/transaction.rs]: nit: you can use {:#x} to get the 0x prefix (cf. https://doc.rust-lang.org/nightly/core/fmt/trait.LowerHex.html)

**Review by fleupold — APPROVED** [review]
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia0.giphy.com%2Fmedia%2FyvzK4m2EoIKs9K0GoF%2Fgiphy.gif&token=chain" data-gitmeme-token="chain"></a>

File: src/transaction.rs (hunk 2 above)
Comment (nlordell) [reply src/transaction.rs]: Woah... TIL!

**Pull request merged**